### ISSUE TITLE
Way of organising channels

### ISSUE THREAD

**Issue by matthias-brun** [RocketChat/feature-requests#637]

**Comment by simon-doerflinger** [comment]
Are there any news to this topic? It would be a great Feature and could help us a lot 😁

**Comment by jaenulton** [comment]
Collapsible would be a must. The feature request is nearly 4 years old, any update yet?

**Comment by neylwalecki** [comment]
+1

**Comment by Balecan** [comment]
I agree with previous post. This feature is very interesting to organize our teams channel. Some ppl know equivalent feature in other tools and it could be to find the same thing in Rocket Chat to help improve is usage. Ty

**Comment by clemoan** [comment]
+1

**Comment by BrennenHodge** [comment]
This feature is even more important with the increase of remote teams. 

I vote we bump this up to higher priority.

**Comment by nemeth-it** [comment]
+1

**Comment by avatar1024** [comment]
+1 for this too! It would be very useful. It also feels that we should be able to sort discussions as sub channels (since they are created within a channel). So it seems like a three level tree would be best:
Room: top level which is just used to group channels > can give it a description but not for people to post messages
Channels: top layer for people to chat
Discussions: specific topic within a channel > effectively sub channels

**Comment by charliehelmijr** [comment]
> +1 for this too! It would be very useful. It also feels that we should be able to sort discussions as sub channels (since they are created within a channel). So it seems like a three level tree would be best:
> Room: top level which is just used to group channels > can give it a description but not for people to post messages
> Channels: top layer for people to chat
> Discussions: specific topic within a channel > effectively sub channels

This would be perfect.

**Comment by ghost** [comment]
+1

**Comment by GrzegorzBobrowicz** [comment]
+1

**Comment by zenithphysics** [comment]
+100 for this feature. Its really useful and would make things so much more efficient. 

Check : Telegram's folder structure also. Everything is moving towards better organization.

**Comment by tetefolle** [comment]
+1

**Comment by Tejeev** [comment]
+1
Easily the most desirable improvement to RocketChat for me

**Comment by Creaous** [comment]
+1

**Comment by tobias-mierzwa** [comment]
+1

**Comment by robbyoconnor** [comment]
THis would be cool -- I could split out for example Google Summer of Code by year!

**Comment by samdup** [comment]
+1 - lack of channel organization makes the communication tool get more and more cumbersome and intimidating with time. I'd love to see this feature (ideally for admins) with Rocket.chat! 💖

Here's what some competitors have done:
- [Slack has this as a "sections" feature](https://slack.com/help/articles/360043207674-Organize-your-sidebar-with-custom-sections#organize-your-conversations). On enterprise-grid you can control the sections for all users.
- [Discord has categories and subchannels](https://support.discord.com/hc/en-us/articles/115001580171-Channel-Categories-101) that admins can control
- [Mattermost enables users to organize channels](https://community-daily.mattermost.com/core/channels/ux-channel-sidebar), but the admin can't organize it for all users.

**Comment by hawatius** [comment]
+1

**Comment by ankar84** [comment]
Other issues related to that theme
https://github.com/RocketChat/feature-requests/issues/660
https://github.com/RocketChat/Rocket.Chat/issues/18803
https://github.com/RocketChat/feature-requests/issues/451

**Comment by v360-eduardo-marques** [comment]
+1

**Comment by aakash-gitdev** [comment]
I think this issue still persists. The list in the channel section should be made collapsible. Are there any updates regarding this?

**Comment by vesperk** [comment]
+1

**Comment by Meibisu** [comment]
+1

**Comment by adestis-ds** [comment]
A MUST HAVE Feature! @samdup wrote already a year ago that competitors DO HAVE this feature but no activity here...

**Comment by adestis-ds** [comment]
I vote for including this into the mid-term milestone!

**Comment by ArtemT** [comment]
@clevertree, have you ever heard of [Hanlon's razor](https://en.wikipedia.org/wiki/Hanlon%27s_razor)?

> Never attribute to malice that which is adequately explained by stupidity

**Comment by b90g** [comment]
the last three comments are not helping. please check if you want to have such environment..

**Comment by Meibisu** [comment]
> the last three comments are not helping. please check if you want to have such environment..

This feature is required since 2016, and still no reply from developers...

**Comment by robbyoconnor** [comment]
> This feature is required since 2016, and still no reply from developers...

@Meibisu Why not implement it yourself? This is open-source, don't demand things.

**Comment by subspecs** [comment]
> > This feature is required since 2016, and still no reply from developers...
> 
> @Meibisu Why not implement it yourself? This is open-source, don't demand things.

Well we're sorry that we asked such a basic necessity of the poor poor developers.

**Comment by Meibisu** [comment]
> > This feature is required since 2016, and still no reply from developers...
> 
> @Meibisu Why not implement it yourself? This is open-source, don't demand things.

Because i use their enterprise license. And dont want to implement things and change them with every update. That's developers task and very important improvement which available in many others messendgers.

**Comment by jimintz** [comment]
Obligatory 2023 bump

**Comment by T1mors** [comment]
Is there an update on this? This feature seems to be very popular and we would also like to organise channels ourselves

**Comment by ezleszken** [comment]
Why is this feature still not added?

**Comment by RuBAN-GT** [comment]
Any updates?

**Comment by adestis-ds** [comment]
It would be very nice if developers could comment the state here. It is not very helpful to see +1 from many customers for more than 7 years and no reply from developers. We are an enterprise customer and really need this feature (so a +1 from us as well).

**Comment by ankar84** [comment]
> We are an enterprise customer and really need this feature (so a +1 from us as well).

Please create a paid service desk ticket to show importance of that feature fir you.

**Comment by clevertree** [comment]
$$$ is in charge. Open source is dead.

**Comment by neylwalecki** [comment]
@clevertree @adestis-ds and all

They're problably strugling to make profit and there's to many open source users, that's bad for their business. So they started to remove and limit features from the open source version and now are disabling push notifications and other things from who doesn't upgrade to the most recent versions (which has the new limitations imposed).

They're doing this at a slow pace and hiding it among other announcements, so most users problably have no knowledge of this until they need to use something that was there before and now shows a message about upgrade to the paid plan.

Things that were removed or limited:

- No more home screen custom message/HTML
- No more read receipt/info on messages
- You can't create new permissions profile (type of user), but if you created before they continue to work (otherwise users would get very very mad)
- You can't activate more than 5 apps from the marketplace (and some popular free apps now are paid, like polls, for example)
- They ended the benefit of doubling the push notification limit for who donated 10USD montly here in their GitHub (but are not enforcing the limits yet as they need to offer a paid option to buy the push notification extra amount first)

Problably there's more things, specially for who uses omnichannel feature (not my case), these is what I noticedin these last year or so.

Rocketchat is a great app and I really admire their work. I'm from Brazil and I'm proud to have a brazilian startup like Rocketchat being successful. But they should offer more tiers/options for who runs it on-premises and don't need omnichannel and other things.

**Comment by SerPenTargo** [comment]
It's very strange that RC still doesn't have the functionality to group the contacts window by category or department. This is a much needed functionality.

**Comment by Gitoffomalawn** [comment]
I feel this issue needs another necrobump to stay on the radar...

**Comment by vorpentine** [comment]
Grouping channels would be amazing

**Comment by warlock9k** [comment]
Another option that seems pretty simple to implement is to add an additional “Label” field to the channel property. And add a “Group by label” option to the appearance. This would allow different channels to be grouped together.
Ideally, labels should be different for each user, but if this is difficult to implement, you can make them available for editing to the channel owner.

**Comment by zigmasdirigeant** [comment]
Oh wow, I tough it was me not being able to see a way to do this. So, it's still not viable?

**Comment by kobzar** [comment]
Any updates?

**Comment by PlasmaDev5** [comment]
Bump on this. I have many channels in my server and this is prior to bringing the team in to create discussions and direct messages. We really need a way to group chats into folder or sup groups. 
I already have all chats belonging to a team so simply having them be a drop down under the team would be a huge cleanup. Same goes with having discussions down down under there owning channel

**Comment by lrnd1** [comment]
As this feature wasn't implemented in 10 years, having "planned" tag since 2019, it seems unlikely to come despite so many people really interested in it. 
For me personally it's a dealbreaker. I prefer more future-proof solutions where I can keep my environment clean and organized. I'll probably revisit RC in the future if this ever gets sorted out because I really like the main concept and other features.

**If anyone knows a custom solution or community workaround for adding this feature, be so kind and share it!** - I could not find one.
